src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health) {
+    this.name = name;
+    this.health = health || 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(name) {
+    if (name.hidden || name instanceof Carnivore) {
```

Comment:
try to use more relevant naming of variables  - name it's not the best use beast or animal instead. Adl this IF is is a bit strange am I right?  Maybe it's better to check if the animal is Herbivore and hidden or note to minus health

Reply:
fix @GarryRocksman comment